dbName follows dependencies.indexedDB

mcshaz opened the pull request:
setting Dexie.dependencies.indexedDB = indexedDB;
now updates the databaseNamesEnumerator so that the table of names is in the same database provider

addresses point 1 of issue #816

still has not addressed including the IDBFactory when instantiating a new Dexie object 

all tests passing

dfahlander commented:
Thanks for this PR! I see that tests does not currently pass. Just for your info, I'm currently having issues with travis/karms/browserstack also for the master branch but haven't nailed down the issue yet. Probably due to external causes (changes in the travis or browserstack services).

I will keep trying to find the reason for the travis issues.

mcshaz commented:
thank you - it was passing until I did something - i'll try and nail it down

dfahlander commented:
Just want to say, it might be problems with travis/browserstack, since I'm having issues in master branch as well, and that was before your PR.

mcshaz commented:
can you please let me know when the travis/browserstack issue is fixed, and I'll have another look - thank you.

dfahlander commented:
Seems the pure dexie tests pass, so that's good. Probably it's only the travis-issue that makes the addon-tests fail. I'll let you know when the travis issues are fixed, but I can't promise when that will be, but as long as it's not fixed, we're blocked on any PRs.

dfahlander commented:
@mcshaz you can also subscribe to #821 to get notified about the travis-issue.

mcshaz commented:
this is just being closed and reopened to rerun the Travis CLI build now that issue is fixed

mcshaz commented:
passing, and passes when I run tests in IE. Bizarrely, when I first ran the tests in IE, it failed for a single test with:
```
    × modify-with-where(issue-594)
      IE 11.0.0 (Windows 7.0.0)
    There should be 3 Davids
    Expected: 3
    Actual: 2
 ```
but has passed all tests on subsequent occasions. of note the original issue seems to be safari, and I am not sure if the test is truly encapsulated.

to be clear on what this test allows:
```typescript
export class DrugsDBLocal extends Dexie implements IDrugDB {
    constructor(indexedDb?: IDBFactory,
                      dbKeyRange?: typeof IDBKeyRange) {
        if (indexedDb !== void 0) {
            Dexie.dependencies.indexedDB = indexedDb as IDBFactory;
        }
        if (dbKeyRange !== void 0) {
            Dexie.dependencies.IDBKeyRange = dbKeyRange;
        }
        super('DrugData')
```

dfahlander commented:
Good that the tests succeeds. Internet Explorer is shaky. The browser fail sporadically on the unit tests and always has.

What I found in the review was the try..catch being removed. The reason for the try..catch can be found in issue #614, #617 and PR #619.

mcshaz commented:
No worries – I’ll put it back in.

From: David Fahlander [mailto:notifications@github.com]
Sent: Tuesday, 14 May 2019 10:09 p.m.
To: dfahlander/Dexie.js
Cc: Brent McSharry (ADHB); State change
Subject: Re: [dfahlander/Dexie.js] dbName follows dependencies.indexedDB (#820)


Good that the tests succeeds. Internet Explorer is shaky. The browser fail sporadically on the unit tests and always has.

What I found in the review was the try..catch being removed. The reason for the try..catch can be found in issue #614<https://scanmail.trustwave.com/?c=7264&d=yZPa3LjmU_19oasoOMOVCHP9qluNQC74eR5HZekRJQ&u=https%3a%2f%2fgithub%2ecom%2fdfahlander%2fDexie%2ejs%2fissues%2f614>, #617<https://scanmail.trustwave.com/?c=7264&d=yZPa3LjmU_19oasoOMOVCHP9qluNQC74eUhGMu0aKQ&u=https%3a%2f%2fgithub%2ecom%2fdfahlander%2fDexie%2ejs%2fissues%2f617> and PR #619<https://scanmail.trustwave.com/?c=7264&d=yZPa3LjmU_19oasoOMOVCHP9qluNQC74eRhDMu4aeQ&u=https%3a%2f%2fgithub%2ecom%2fdfahlander%2fDexie%2ejs%2fpull%2f619>.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub<https://scanmail.trustwave.com/?c=7264&d=yZPa3LjmU_19oasoOMOVCHP9qluNQC74eR4TPe0bKQ&u=https%3a%2f%2fgithub%2ecom%2fdfahlander%2fDexie%2ejs%2fpull%2f820%3femail%5fsource%3dnotifications%26email%5ftoken%3dAASXZAFKVB3DF7WKHQQNBSDPVKFUJA5CNFSM4HCOF3A2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVK77OA%23issuecomment-492175288>, or mute the thread<https://scanmail.trustwave.com/?c=7264&d=yZPa3LjmU_19oasoOMOVCHP9qluNQC74eUpHZ71PKw&u=https%3a%2f%2fgithub%2ecom%2fnotifications%2funsubscribe-auth%2fAASXZACEW5K4BACMIU6SAG3PVKFUJANCNFSM4HCOF3AQ>.

mcshaz commented:
the try catch is back in the original state. having said that, I think it could be potentially improved further - I am as far from an expert on indexedDB as could be, but the error messages & stack trace I have seen are quite unambiguous in Firefox at least - the error is because of insufficient DB permissions. I am not as convinced a null reference would give just as much information. If you want to wait until calling db open, what I would propose is something along the lines of:

```typescript
    public constructor(args?: IDexieDependencies){
        if (args !== void 0){
            this.IDBKeyRange = args.IDBKeyRange;
            this.indexedDB = args.indexedDB;
            this.pIDBError = args.iDBError;
        }
    }
    public get indexedDB(){
        if (this.pIDBError) { throw this.pIDBError}
        return this.pIndexedDB;
    }
... 'dexie-static-props
  dependencies: new DexieDependencies((() => {
    try {
      return {
        indexedDB: (_global.indexedDB || _global.mozIndexedDB || _global.webkitIndexedDB || _global.msIndexedDB) as IDBFactory,
        IDBKeyRange: (_global.IDBKeyRange || _global.webkitIDBKeyRange) as IDBKeyRange
      }
    } catch (err) {
      return { indexedDB: null, IDBKeyRange: null, iDBError: err }
    }
  })()),
```

I have not instituted this as this would be  a breaking change

mcshaz commented:
just bumping this as so far no response since making the changes a month ago, and wondering if it has been forgotten  - as per your comment, I had added back in the enclosed function & try/catch block to the PR, + added a new comment explaining why the try/catch block. 

On a separate branch (not in this PR) I hold onto the original browser error in order to rethrow when first accessing Dexie

dfahlander commented:
I haven't got any time for getting a full understanding of the changes. I've had a lot to do, and this PR comes with new logic that is not obvious so i would need to spend more time on it.

mcshaz commented:
No worries, I had just thought it might have been forgotten about.

dfahlander commented:
Replaced by #1194
